badRegexp: tests fail with go1.19

go1.18

> --- PASS: TestCheckers/badRegexp (0.06s)

go1.19

> --- FAIL: TestCheckers/badRegexp (0.04s)

```
linttest.go:170: testdata/badRegexp/positive_tests.go:81: unexpected warn: `_` intersects with `\w` in [\w!#$%&'*+/=?^_`{|}~-]
linttest.go:170: testdata/badRegexp/positive_tests.go:151: unexpected warn: `=-_` intersects with `=` in [/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》]
linttest.go:170: testdata/badRegexp/positive_tests.go:169: unexpected warn: `\w` intersects with `\d` in [!\w\d\.\+\-]
linttest.go:170: testdata/badRegexp/positive_tests.go:173: unexpected warn: `\w` intersects with `\d` in [_=\w\d\.&;]
linttest.go:200: testdata/badRegexp/positive_tests.go:169: unmatched ``\d` intersects with `\w` in [!\w\d\.\+\-]`
linttest.go:200: testdata/badRegexp/positive_tests.go:151: unmatched ``=` intersects with `=-_` in [/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》]`
linttest.go:200: testdata/badRegexp/positive_tests.go:81: unmatched ``\w` intersects with `_` in [\w!#$%&'*+/=?^_`{|}~-]`
linttest.go:200: testdata/badRegexp/positive_tests.go:173: unmatched ``\d` intersects with `\w` in [_=\w\d\.&;]`
```

--- a/checkers/badRegexp_checker.go
+++ b/checkers/badRegexp_checker.go
@@ -365,7 +365,7 @@ func (c *badRegexpChecker) checkCharClassDups(cc syntax.Expr) {
 	}
 
 	// 2. Sort ranges, O(nlogn).
-	sort.Slice(ranges, func(i, j int) bool {
+	sort.SliceStable(ranges, func(i, j int) bool {
 		return ranges[i].low < ranges[j].low
 	})
 

--- a/checkers/testdata/badRegexp/positive_tests.go
+++ b/checkers/testdata/badRegexp/positive_tests.go
@@ -147,7 +147,7 @@ func suspiciousAltAnchor() {
 
 func badRegexps() {
 	/*! suspicious char range `=-_` in [/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》] */
-	/*! `=` intersects with `=-_` in [/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》] */
+	/*! `=-_` intersects with `=` in [/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》] */
 	regexp.MustCompile(`[/.@!~#$%^&*:";?\\+=-_,{}\[\]<>！￥…（）—=、“”：；？。，《》]`)
 
 	/*! `e` is duplicated in [com|org|edu|net] */
@@ -165,10 +165,10 @@ func badRegexps() {
 	/*! `_` is duplicated in [a-zA-Z\.\-_0-9_] */
 	regexp.MustCompile(`^[a-zA-Z\.\-_0-9_]+$`)
 
-	/*! `\d` intersects with `\w` in [!\w\d\.\+\-] */
+	/*! `\w` intersects with `\d` in [!\w\d\.\+\-] */
 	regexp.MustCompile(`^https?://itunes.apple.com/(?:(\w+)/)?app/(?:[!\w\d\.\+\-]+/)?id(\d+)`)
 
-	/*! `\d` intersects with `\w` in [_=\w\d\.&;] */
+	/*! `\w` intersects with `\d` in [_=\w\d\.&;] */
 	/*! `\w` intersects with `\d` in [_\w\d\.] */
 	regexp.MustCompile(`^https?://play.google.com/store/apps/details\?(?:[_=\w\d\.&;]*[;|&])?id=([_\w\d\.]+)`)
 

